Engine/TestStepRun.cs

```diff
@@ -381,7 +381,9 @@ internal void AfterRun(ITestStep step)
             {
                 if (member.HasAttribute<ResultAttribute>())
                 {
-                    if (member.TypeDescriptor.IsPrimitive())
+                    var td = member.TypeDescriptor.AsTypeData();
+                    
+                    if (td != null && (td.IsNumeric || td.IsString))
```

Comment:
I'm not sure exactly what is included in `IsPrimitive` compared to `IsNumeric`. Is a boolean value numeric? 

To be safe, I think we should relax the check to allow all primitives in addition to strings.